### ISSUE TITLE
`Error: Initial console modes not set` on Windows

### ISSUE THREAD

**Issue by fujiapple852** [crossterm-rs/crossterm#660]
**Describe the bug**
Receive the error `Error: Initial console modes not set` when attempting to test [trippy](https://github.com/fujiapple852/trippy) on Windows.  

The error seems to originate from [`windows.rs`](https://github.com/crossterm-rs/crossterm/blob/master/src/event/sys/windows.rs#L34) which states:

> Returns the original console color, make sure to call `init_console_color` before calling this function. Otherwise this function will panic.

So presumably `init_console_color` isn't being called.  I'm unsure if this is something application code should be invoking (on Windows) or whether it is an internal API within `crossterm`.

When I run the tool I observe no output at all and it is only when I exit the tool that the screen renders and the error occurs.

**To Reproduce**
See the steps outlined [here](https://github.com/fujiapple852/trippy/issues/98).

**OS**
Windows 10 from AWS WorkSpaces, bundle "Standard with Windows 10 (Server 2019 based) (WSP)"

**Terminal/Console**
Observed on Windows console (cmd.exe) and also in Powershell

**Comment by TimonPost** [comment]
This error is given when `init_original_console_mode` is not called. This is called when you enable mouse capture (`EnableMouseCaputre`). It should only give this error if your calling `DiableMouseCapture` while you've never enabled it. Could that be right?

**Comment by fujiapple852** [comment]
@TimonPost that looks to be the case:

https://github.com/fujiapple852/trippy/blob/master/src/frontend.rs#L181

**Comment by fujiapple852** [comment]
@TimonPost I've rested and can confirm that the error no longer occurs and so this can be closed, thanks again.

There is still the other issue I mentioned, that the app doesn't display _anything_ at all (completely blank terminal screen) after it starts until the application exits, at which point the final state of the app is displayed.  

In general, should I expect `crossterm` to work on the OS/terminal combinations I mentioned above?  If so I'll raise this as a separate issue (I tried the [`tui-rs` demo app](https://github.com/fdehau/tui-rs/tree/master/examples/demo) and observed the same problem which makes me think I'm using a OS/terminal combination that isn't supported).  I don't have a local Windows machine to test with sadly.

**Comment by TimonPost** [comment]
What terminal modes are you in? Did you enable raw modes/alternate screen?

**Comment by fujiapple852** [comment]
@TimonPost to rule out all other variables I just retested with the crossterm [interactive-demo](https://github.com/crossterm-rs/crossterm/tree/master/examples/interactive-demo).

I spun up a fresh AWS WorkSpaces machine (bundle `Performance with Windows 10 (Server 2019 based) (WSP)`), installed `git`, `rust` and the [MS build tools ](https://visualstudio.microsoft.com/downloads/?q=build+tools#build-tools-for-visual-studio-2022) and the proceeded to build and run the demo.

Here is a video of what I see, which is consistent with what I saw from `tui-rs` and my own app:

![crossterm](https://user-images.githubusercontent.com/36075719/167079694-a628b281-a565-467c-8eb4-0937c6497ac0.gif)

Specifically, when the app starts, the screen clears and remains blank until you exit the app (I pressed 'q' after a few seconds) at which point the final state of the screen is shown.

I see the same thing for both Powershell and the `cmd.exe`.
